core/modules/replica/FileIngestApp.cc

```diff
@@ -0,0 +1,291 @@
+        ++lineNum;
+
+        // Fields to be parsed:
+        // <worker-host> <worker-port> <transaction-id> <table> <type> <infile>
+        regex const re("^(.+)[ ]+([0-9]+)[ ]+([0-9]+)[ ]+(.+)[ ]+(P|R)[ ]+(.+)$", regex::extended);
+        smatch match;
+        if (not regex_search(line, match, re) or match.size() != 7) {
+            throw invalid_argument(
+                    context
+                    + "Failed to parse the content of file " + _fileListName + ":" + to_string(lineNum)
+                    + ", expected fields: <worker-host> <worker-port> <transaction-id> <table> <type> <infile>.");
+        }
+
+        FileIngestSpec file;
+        file.workerHost = match[1].str();
+
+        auto const workerPort = stoul(match[2].str());
+        if (workerPort < 1 or workerPort > numeric_limits<uint16_t>::max()) {
+            throw invalid_argument(
+                    context
+                    + "Failed to parse the content of file " + _fileListName + ":" + to_string(lineNum)
+                    + ", a value " + to_string(workerPort) + " of <worker-port> is not in a range of 1-"
+                    + to_string(numeric_limits<uint16_t>::max()) + ".");
+        }
+        file.workerPort = (uint16_t)workerPort;
+
+        auto const transactionId = stoul(match[3].str());
+        if (transactionId > numeric_limits<uint32_t>::max()) {
+            throw invalid_argument(
+                    context
+                    + "Failed to parse the content of file " + _fileListName + ":" + to_string(lineNum)
+                    + ", a value " + to_string(transactionId) + " of <transaction-id> is not in a range of 0-"
+                    + to_string(numeric_limits<uint32_t>::max()) + ".");
+        }
+        file.transactionId = (uint32_t)transactionId;
+        file.tableName = match[4].str();
+        file.tableType = match[5].str();
+        file.inFileName = match[6].str();
+        
+        files.push_back(file);
+    }
+    return files;
+}
+
+
+void FileIngestApp::_ingest(FileIngestSpec const& file) const {
+    string const context = "FileIngestApp::" + string(__func__) + "  ";
+
+    // Analyze the file to make sure it's a regular file, and it can be read.
+
+    fs::path const path = file.inFileName;
+    boost::system::error_code ec;
+    fs::file_status const status = fs::status(path, ec);
+    if (ec.value() != 0) {
+        throw invalid_argument(context + "file doesn't exist: " + path.string());
+    }
+    if (not fs::is_regular(status)) {
+        throw invalid_argument(context + "not a regular file: " + path.string());
+    }
+    
+    // For partitioned tables analyze file name and extract a chunk number and
+    // the 'overlap' attribute
+    unsigned int chunk = 0;
+    bool isOverlap = false;
+
+    if (file.tableType == "P") {
+        string const filename = fs::absolute(path).filename().string();
+        regex const re("^chunk_([0-9]+)(_overlap)?\\.txt$", regex::extended);
```

Comment:
Some unit testing would make me more comfortable about things in general (including maintainability and poor future developers who must do it), and especially about knowing these regex strings working the way they're supposed to.